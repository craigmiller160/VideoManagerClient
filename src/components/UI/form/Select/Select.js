import React from 'react';
import PropTypes from 'prop-types';
import ReactSelect from 'react-select';
import { FormGroup, Label } from 'reactstrap';
import newid from '../../../../utils/newid';
import formStyles from '../FormStyles.scss';
import createField from "../createField";

export const SelectComponent = (props) => {
    const {
        label,
        input,
        name,
        options,
        multi
    } = props;

    console.log('Multi', multi); // TODO delete this

    const id = newid();
    return (
        <FormGroup className={ formStyles['input-label'] }>
            <Label
                for={ id }
            >
                { label }
            </Label>
            <ReactSelect
                id={ id }
                { ...input }
                name={ name }
                onBlur={ () => input.onBlur(input.value) }
                options={ options }
                isSearchable
                isClearable
                isMulti={ multi }
            />
        </FormGroup>
    );
};

SelectComponent.propTypes = {
    label: PropTypes.string.isRequired,
    options: PropTypes.array.isRequired,
    input: PropTypes.shape({
        name: PropTypes.string
    }),
    multi: PropTypes.bool
};

SelectComponent.defaultProps = {
    options: [],
    multi: false
};

const Field = createField(SelectComponent);
Field.propTypes = {
    label: PropTypes.string,
    name: PropTypes.string,
    multi: PropTypes.bool
};
export default Field;