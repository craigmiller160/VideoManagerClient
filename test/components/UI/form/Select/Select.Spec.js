import React from 'react';
import Select from '../../../../../src/components/UI/form/Select/Select';
import { shallow } from 'enzyme';

describe('Select', () => {
    const props = {
        label: 'My Select',
        name: 'MySelect',
        options: [
            { label: 'Opt1', value: 'opt1' },
            { label: 'Opt2', value: 'opt2' }
        ]
    };

    it('should render correctly', () => {
        const component = shallow(
            <Select { ...props } />
        );
        expect(component).toMatchSnapshot();
    });

    it('should handle onChange', () => {
        let value = 'InitialValue';
        props.onChange = (newValue) => value = newValue;

        const component = shallow(
            <Select { ...props } />
        );

        const result = component.find('[name="MySelect"]');
        expect(result.length).toEqual(1);

        result.simulate('change', { label: 'Opt2', value: 'opt2' });
        expect(value).toEqual({ label: 'Opt2', value: 'opt2' });
    });
});