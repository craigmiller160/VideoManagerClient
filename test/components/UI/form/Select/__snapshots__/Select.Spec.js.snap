// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select should render correctly 1`] = `
<Field
  component={[Function]}
  label="My Select"
  name="MySelect"
  options={
    Array [
      Object {
        "label": "Opt1",
        "value": "opt1",
      },
      Object {
        "label": "Opt2",
        "value": "opt2",
      },
    ]
  }
/>
`;
